docs/rfcs/1020-pki-support.md

```diff
@@ -0,0 +1,291 @@
+They would like that we use their certificate to access Parsec, so the user does not have to use a password or rely on the keyring of the OS.
+
+There is also an opportunity to rely on that PKI to join an organization.
+
+Put in a simpler term access Parsec using a Smartcard.
+
+### Terms:
+
+- **PKI**: Public Key Infrastructure
+- **Smartcard** or **Token**:
+  A tangible device that is similar to an USB stick (that store data).
+  That device is used to securely store private key, secrets and is able to perform cryptographic operations
+- **Submitter**: A user/device submitting a request to join an organization using the PKI feature.
+- **Accepter**: A user/device admin of an organization in charge of accepting or not the join request.
+
+## Goals and Non-Goals
+
+Some prior work was already done to add some primitives (sign, verify, encrypt, decrypt) to work with the windows certificate store under the `libparsec_platform_pki`.
+
+The goals are to define the higher elements to make the PKI work with Parsec for the invitation process, those higher elements are:
+
+- [`pki_enrollment_select_certificate`]
+- [`pki_enrollment_sign_payload`]
+- [`pki_enrollment_create_local_pending`]
+- [`pki_enrollment_load_peer_certificate`]
+- [`pki_enrollment_load_submit_payload`]
+- [`pki_enrollment_load_accept_payload`]
+- [`pki_enrollment_load_local_pending_secret_part`]
+
+[`pki_enrollment_select_certificate`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L217-L251
+[`pki_enrollment_sign_payload`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L254-L276
+[`pki_enrollment_create_local_pending`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L279-L333
+[`pki_enrollment_load_peer_certificate`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L336-L355
+[`pki_enrollment_load_submit_payload`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L385-L406
+[`pki_enrollment_load_accept_payload`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L409-L430
+[`pki_enrollment_load_local_pending_secret_part`]: https://github.com/Scille/parsec-extensions/blob/5f0ae5ac44d23d775d92e241b546fc0cac327a0a/parsec_ext/smartcard/__init__.py#L433-L473
+
+## Design
+
+Here is a possible user flow of Alice joining the organization CoolOrg using the PKI to enable the async enrollment feature:
+
+ 1. Alice wants to join CoolOrg organization.
+ 2. Alice already has a PKI configured on its system and the link to submit the join request.
```

Comment:
What does it means to have a "PKI configured on its system"? In addition to having access to its physical smartcard, what Alice needs to configure?